### PR TITLE
Fix forward sync explicit index ref

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Fix forward sync explicit index ref

## Motivation and Context

There was no check if length of blocks received was 0 in one use of the blocks vec, this PR fixes that. 

## Types of changes

* [x Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/base_node/states/forward_block_sync.rs [hunk 1]
```diff
@@ -237,13 +237,12 @@ async fn download_blocks<B: BlockchainBackend + 'static>(
...
-                info.tip_height = Some(blocks[blocks.len() - 1].block().header.height);
-                info.local_height = Some(blocks[0].block().header.height);
-            }
-            shared.publish_event_info().await;
             for i in 0..blocks.len() {
+                if let StatusInfo::BlockSync(ref mut info) = shared.info {
+                    info.tip_height = Some(blocks[blocks.len() - 1].block().header.height);
+                    info.local_height = Some(blocks[0].block().header.height + (i as u64));
+                }
+                shared.publish_event_info().await;
```
Comment (sdbondi) [inline base_layer/core/src/base_node/states/forward_block_sync.rs:245]: Did you mean to publish this `blocks.len()` times? - I think it's better to just add the `!blocks.is_empty()` check above the loop

**Review by sdbondi — COMMENTED** [review]
One comment

File: base_layer/core/src/base_node/states/forward_block_sync.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/core/src/base_node/states/forward_block_sync.rs:245]: Wel it depends on the accuracy you want the status to be shown. If you want it to only update once per batch of blocks updated, then it should be above the for loop. 
If not then you need to publish this channel every block. 

Its easy to update it both ways, I am not particular to one side or the other

Comment (CjS77) [reply base_layer/core/src/base_node/states/forward_block_sync.rs:245]: Agree. Rather just publish the status update once.

Comment (SWvheerden) [reply base_layer/core/src/base_node/states/forward_block_sync.rs:245]: cwl, updated

**Review by sdbondi — APPROVED** [review]
LGTM

**Pull request merged**